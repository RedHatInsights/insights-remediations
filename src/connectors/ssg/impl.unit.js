'use strict';

const fs = require('fs');
const path = require('path');
const impl = require('./impl');
const base = require('../../test');
const { mockRequest } = require('../testUtils');
const request = require('../../util/request');

const REQ = {
    headers: {
        'x-rh-identity': 'identity',
        'x-rh-insights-request-id': 'request-id'
    },
    identity: { type: 'test' },
    user: { username: 'test', account_number: 'test' }
};

describe('ssg impl', function () {
    beforeEach(mockRequest);

    test('returns template', async function () {
        // eslint-disable-next-line security/detect-non-literal-fs-filename
        const data = fs.readFileSync(path.join(__dirname, 'mock', 'standard', 'service_rsyslog_enabled.yml'), 'utf-8');

        const http = base.getSandbox().stub(request, 'run').resolves({
            statusCode: 200,
            body: data,
            headers: {
                'x-ssg-version': 'unit'
            }
        });

        const { template, version } = await impl.getTemplate(REQ, 'rhel7', 'standard', 'sshd_disable_root_login');
        version.should.equal('unit');
        expect(template).toMatchSnapshot();
        http.callCount.should.equal(1);
    });
});
