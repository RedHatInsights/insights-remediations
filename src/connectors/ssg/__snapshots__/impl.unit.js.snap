// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ssg impl returns template 1`] = `
"# platform = multi_platform_all
# complexity = low
# strategy = enable
# reboot = false
# disruption = low
- name: Enable rsyslog Service
  hosts: '@@HOSTS@@'
  become: true
  tags:
    - CCE-80188-6
    - NIST-800-53-AU-12
    - NIST-800-53-AU-4(1)
    - enable_strategy
    - low_complexity
    - low_disruption
    - medium_severity
    - no_reboot_needed
    - service_rsyslog_enabled
  tasks:

    - name: Enable service rsyslog
      block:

        - name: Gather the package facts
          package_facts:
            manager: auto

        - name: Enable service rsyslog
          service:
            name: rsyslog
            enabled: 'yes'
            state: started
          when:
            - '"rsyslog" in ansible_facts.packages'
      when: ansible_virtualization_role != "guest" or ansible_virtualization_type
        != "docker"
"
`;
